### PR TITLE
Validate features - enforce types, keep ints in range

### PR DESCRIPTION
**Pull request by olsen232**
Until now, we don't validate data on commit. Since GPKG doesn't care how many bits an integer has, committing a GPKG working copy can result in the dataset containing integers with too many bits - this would stop you from checking out a postgis working copy.

https://github.com/koordinates/sno/issues/293

## Checklist:

- [x] Have you reviewed your own change?
- [x] Have you included test(s)?
- [x] Have you updated the [changelog](https://github.com/koordinates/sno/blob/master/CHANGELOG.md)?

### CODE REVIEW

**Review by craigds — APPROVED** [review]
nice work 🎉

File: sno/schema.py [hunk 1]
```diff
@@ -451,3 +452,50 @@ def diff_types(self, new_schema):
...
+            if type(value) not in self.EQUIVALENT_PYTHON_TYPES[col_type]:
+                has_violation = True
+                if col.name not in violation_info:
+                    message = f"In column {col.name} value {repr(value)} doesn't match schema type {col_type}"
+                    violation_info[col.name] = message
+            elif col_type == "integer" and "size" in col.extra_type_info:
+                size = col.extra_type_info["size"]
+                if self._signed_bit_length(value) > size:
+                    has_violation = True
+                    if col.name not in violation_info:
```
[the remark this answers is not in the document]
Comment (olsen232) [reply sno/schema.py]: Refactored this whole thing to be clearer - broke it into two methods.

**Comment by olsen232** [comment]
This change does also enforce types themselves (eg str vs int) but I was unaware of how important this was - have updated the documentation and the tests now

**Comment by rcoup** [comment]
@olsen232 any performance impact from this?

**Pull request merged**